[Guide] Configuration profiles for Android

marko-lisica opened the pull request:
Related to:

- #25557

marko-lisica committed:
Update custom-os-settings.md

marko-lisica committed:
Update custom-os-settings.md

marko-lisica committed:
Update custom-os-settings.md

marko-lisica committed:
Update routes.js

mike-j-thomas reviewed (approved):
Looks good to me. I just spotted and fixed one typo.

mike-j-thomas committed:
Update articles/custom-os-settings.md

noahtalerman commented on articles/custom-os-settings.md:
```diff
@@ -62,13 +66,18 @@ In versions older than 4.71.0, Fleet always delivered configuration profiles to
 In the top box, with "Verified," "Verifying," "Pending," and "Failed" statuses, click each status to view a list of hosts:
 
-* **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on Windows and macOS hosts (declarations profiles are verified with a [DDM StatusReport](https://developer.apple.com/documentation/devicemanagement/statusreport)). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings.
+* **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on Windows and macOS hosts (declarations profiles are verified with a [DDM StatusReport](https://developer.apple.com/documentation/devicemanagement/statusreport)). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings. Android hosts are "Verified" af ...
```
```suggestion
* **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on Windows and macOS hosts (declarations profiles are verified with a [DDM StatusReport](https://developer.apple.com/documentation/devicemanagement/statusreport)). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings. Android hosts are "Verified" after Fleet gets a status report via the [Android Management API](TODO) from the host.
```

@marko-lisica is there a spot in the Android API doc where users can see what a status report means / looks like?

marko-lisica committed:
Apply suggestion from @noahtalerman

Co-authored-by: Noah Talerman <47070608+noahtalerman@users.noreply.github.com>

marko-lisica committed:
Apply suggestion from @marko-lisica

marko-lisica committed:
Apply suggestion from @noahtalerman

Co-authored-by: Noah Talerman <47070608+noahtalerman@users.noreply.github.com>

marko-lisica committed:
Apply suggestion from @noahtalerman

Co-authored-by: Noah Talerman <47070608+noahtalerman@users.noreply.github.com>

marko-lisica committed:
Update articles/custom-os-settings.md

Co-authored-by: Noah Talerman <47070608+noahtalerman@users.noreply.github.com>

marko-lisica committed:
update

marko-lisica committed:
Apply suggestion from @marko-lisica

marko-lisica committed:
Add route for example Android profile

marko-lisica committed:
Enhance Android settings documentation with error details

Added information about error messages from the Android Management API regarding non-compliance reasons.

marko-lisica commented on articles/custom-os-settings.md:
```diff
@@ -1,26 +1,29 @@
 
 For Windows hosts, copy this [Windows configuration profile template](https://fleetdm.com/example-windows-profile) and update the profile using any [configuration service providers (CSPs)](https://fleetdm.com/guides/creating-windows-csps) from [Microsoft's MDM protocol](https://learn.microsoft.com/en-us/windows/client-management/mdm/).
 
+For Android hosts, copy this [Android configuration profile template](https://fleetdm.com/learn-more-about/example-android-profile) and update the profile using the options available in [Android Management API](https://developers.google.com/android/management/reference/rest/v1/enterprises.policies#resource:-policy). To learn how, watch [this video](TODO).
```
```suggestion
For Android hosts, copy this [Android configuration profile template](https://fleetdm.com/learn-more-about/example-android-profile) and update the profile using the options available in [Android Management API](https://developers.google.com/android/management/reference/rest/v1/enterprises.policies#resource:-policy). To learn how, watch [this video](https://youtu.be/Jk4Zcb2sR1w).
```

marko-lisica committed:
Apply suggestion from @marko-lisica

rachaelshaw committed:
Merge branch 'docs-v4.75.0' into 25557-guides
